src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -91,6 +91,12 @@ private static function compilePattern(Route $route, $pattern, $isHost)
         $matches = array();
         $pos = 0;
         $defaultSeparator = $isHost ? '.' : '/';
+        $useUnicode = preg_match('//u', $pattern);
+        $needsUnicode = $route->getUnicode() || ($useUnicode && preg_match('/[\x80-\xFF]/', $pattern));
```

Comment:
I don't understand the purpose of this detection logic. Why not just rely on `$route->getUnicode()`?